samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -106,4 +112,25 @@ final ExecutionPlan getExecutionPlan(StreamApplication app) throws Exception {
   final StreamManager getStreamManager() {
     return streamManager;
   }
+
+  /**
+   * Write the execution plan JSON to a file
+   * @param planJson JSON representation of the plan
+   */
+  final void writePlanJsonFile(String planJson) {
+    try {
+      String content = "plan='" + planJson + "'";
+      String binPath = System.getenv(ShellCommandConfig.JOB_BIN_DIR());
```

Comment:
I looked at the code again and have to say it's actually more practical to have the json under the bin folder. Since the plan.html needs a static reference to the path of the json file, and the log folder can be configured to anywhere by the user, it's less error-prone to put the json along with the html under bin folder. Since neither bin or log folder in RM box is exposed in yarn web proxy, the user has to log on to the RM box to get the files in either case. So I guess it should be Okey to have it in bin.